CMakeLists.txt

```diff
@@ -254,13 +273,21 @@ else()
   # include path is fine since we already include THIRD_PARTY_ROOT
   # link path is fine since we already include THIRD_PARTY_ROOT
   # RPATH is fine since we already include THIRD_PARTY_ROOT
-  list(APPEND GRAPPA_LIBS
-    boost_system
-    boost_filesystem
-    boost_unit_test_framework)
+  set(Boost_INCLUDE_DIR ${THIRD_PARTY_ROOT}/include)
+  set(Boost_LIBRARY_DIR ${THIRD_PARTY_ROOT}/lib)
   list(APPEND tool_list third-party-boost)
 endif()
 
+# we don't use ${Boost_LIBRARIES} because we want to link the
+# following two libraries statically for all targets, and the unit
+# test library dynamically, but only on test targets.
+list(APPEND GRAPPA_LIBS
+  libboost_system.a
+  libboost_filesystem.a)
```

Comment:
what if we found the multi-threaded version of Boost? Won't we need `libboost_system-mt.a`?